### PR TITLE
Fix #116

### LINKED ISSUE #116: [Bug]: AmpycloudError: Cloud base calculation got an empty array.Maybe check lookback percentage (is set to 100)
### Is there already an issue for this?

- [X] I found no existing issue for this bug.

### Describe the bug

A code user reported receiving the following error:
```
AmpycloudError: Cloud base calculation got an empty array.Maybe check lookback percentage (is set to 100)
```

The issue could be tracked to the DataFrame being fed to ampycloud, that contains something like:
```
#id, dt, alt, type, ceilo
…
177,-30.0,1020.0,1,ceiloX
178,-30.0,,0,ceiloX
179,-30.0,,0,ceiloX
```
All lines are correct on their own, but l.178 and l.179 (no cloud base detected at dt=-30) are directly inconsistent with l. 177 (1 cloud base detected at dt=-30). Lines 178 and 179 are also redundant.

Currently, the function ```check_data_consistency()``` in ```src/ampycloud/utils.py``` does not check for these elements (duplicated hits or inconsistent hits), so that the DataFrame is fed to the codem that eventualy returns a meaningless error.

The solution is to enhance ```check_data_consistency()``` to cover these cases, and issue a meaningful error to inform the user.

### Screenshot

_No response_

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Environment

```markdown
The problem was seen is both ampycloud v1.0.0 and v2.0.0.dev.
```

### Code of Conduct

- [X] I agree to follow the ampycloud Code of Conduct.

### PR DESCRIPTION
**Pull request by fpavogt**
**Description:**

This PR implements a few additional consistency checks for the input data, with the goal to provide the user with (more) meaningful error messages.

**Error(s) fixed:**

Fixes #116

**Checklists**:
- [x] New code includes dedicated tests.
- [x] New code has been linted.
- [x] New code follows the project's style.
- [x] New code is compatible with the 3-Clause BSD license.
- [x] CHANGELOG has been updated.
- [ ] AUTHORS has been updated.
- [ ] Copyright years in module docstrings have been updated.

### CODE REVIEW

**Comment by fpavogt** [comment]
Hello @loforest and @regDaniel, 

Here's a proposition to address #116. This should help improve user experience with better error messages. There are no changes to the scientific behavior of the code.

**Comment by fpavogt** [comment]
@regDaniel: as agreed, I'll wait for your review of this PR before merging #119 into this branch.

File: src/ampycloud/utils/utils.py
Comment (regDaniel) [inline src/ampycloud/utils/utils.py:177]:
```suggestion
   for hit_type in [0, -1]:
        data_hit_type = data[data['type'] == hit_type][['dt', 'ceilo']]
        data_other = data[data['type'] != hit_type][['dt', 'ceilo']]
        common_timestamps = set(data_hit_type['dt']).intersection(set(data_other['dt']))
        if len(common_timestamps) > 0:
            raise AmpycloudError(
                f'Inconsistent input data (simultaneous type {hit_type} and !{hit_type}) '
                f'for the following timestamps:\n {common_timestamps}'
            )
```

File: src/ampycloud/utils/utils.py
Comment (regDaniel) [inline src/ampycloud/utils/utils.py:167]: I wanted to make sure to understand what is being done here and tried to reproduce. For me, the suggestion below is a bit more readable as I'm not so fluent with the merge command. But it's likely a matter of taste and I leave the choice to you @fpavogt .

**Review by regDaniel — APPROVED** [review]
Hi @fpavogt thanks for your work! This will further help users to make sure they have clean input data. See my comment for an alternative formulation, but it was really more about me trying to understand the code.

File: src/ampycloud/utils/utils.py
Comment (fpavogt) [reply src/ampycloud/utils/utils.py:177]: Your change is not equivalent to the original suggestion, because you no longer account for the ceilometer name in the "intersection". I.e. it is perfectly fine to have a VV hit with ceilo A and no hit with ceilo B.

I have added an additional test for this case, such that your proposed change will no longer pass the tests ;)

File: src/ampycloud/utils/utils.py
Comment (fpavogt) [reply src/ampycloud/utils/utils.py:167]: See comment below - proposition is now equivalent to original code.

**Pull request merged**
